Examples/Python/tests/test_examples.py

```diff
@@ -688,36 +688,36 @@ def test_particle_gun(tmp_path, assert_root_hash):
 @pytest.mark.odd
 @pytest.mark.skipif(not dd4hepEnabled, reason="DD4hep not set up")
 def test_material_mapping(material_recording, tmp_path, assert_root_hash):
+    from material_mapping import runMaterialMapping
+    from material_validation import runMaterialValidation
+
     map_file = tmp_path / "material-map_tracks.root"
     assert not map_file.exists()
 
-    s = Sequencer(numThreads=1)
-
     odd_dir = getOpenDataDetectorDirectory()
     config = acts.MaterialMapJsonConverter.Config()
     mdecorator = acts.JsonMaterialDecorator(
         level=acts.logging.INFO,
         rConfig=config,
         jFileName=str(odd_dir / "config/odd-material-mapping-config.json"),
     )
-    detector, trackingGeometry, decorators = getOpenDataDetector(mdecorator)
 
-    from material_mapping import runMaterialMapping
+    s = Sequencer(numThreads=1)
 
-    runMaterialMapping(
-        trackingGeometry,
-        decorators,
-        outputDir=str(tmp_path),
-        inputDir=material_recording,
-        mappingStep=1,
-        s=s,
-    )
+    with getOpenDataDetector(mdecorator) as (detector, trackingGeometry, decorators):
+        runMaterialMapping(
+            trackingGeometry,
+            decorators,
+            outputDir=str(tmp_path),
+            inputDir=material_recording,
+            mappingStep=1,
+            s=s,
+        )
 
-    s.run()
+        s.run()
 
-    # MaterialMapping alg only writes on destruct.
-    # See https://github.com/acts-project/acts/issues/881
-    del s
+        # files are closed in destructors, not great
+        del s
```

Comment:
This might not be true anymore, actually. At least with the `finalize` function we should now have the ability to do this cleanly.